upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go

```diff
@@ -523,9 +533,11 @@ func (v *AutoscalingGroup) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Autos
 		var attachLBRequest *autoscaling.AttachLoadBalancersInput
 		var detachLBRequest *autoscaling.DetachLoadBalancersInput
 		if changes.LoadBalancers != nil {
-			attachLBRequest = &autoscaling.AttachLoadBalancersInput{
-				AutoScalingGroupName: e.Name,
-				LoadBalancerNames:    e.AutoscalingLoadBalancers(),
+			if e != nil && len(e.TargetGroups) > 0 {
```

Comment:
I believe you want this?
```suggestion
			if e != nil && len(e.LoadBalancers) > 0 {
```

Reply:
Thanks for catching this :). Fixed.